polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -1273,45 +1371,109 @@ async fn handle_our_view_change<Context>(
 		for removed in &pruned {
 			gum::debug!(target: LOG_TARGET, relay_parent = ?removed, "Removing relay parent because our view changed.");
 
+			if let Some(block_number) = get_block_number(ctx, *removed).await {
```

Comment:
nit: we could modify `implicit_view.deactivate_leaf` to also return block numbers. It should have them stored already

The backing implicit view could be used in all cases where we are explicitly requesting the block number from ChainApi actually, saving us an extra request for info we already have available

Reply:
Good point.

Reply:
done